sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTextBasedPartitionReader.scala

```diff
@@ -531,8 +532,8 @@ abstract class GpuTextBasedPartitionReader[BUFF <: LineBufferer, FACT <: LineBuf
           isFirstChunk, metrics(GPU_DECODE_TIME))
 
         // parse boolean and numeric columns that were read as strings
-        val castTable = withResource(table) { _ =>
-          castTableToDesiredTypes(table, newReadDataSchema)
+        val castTable = withRetryNoSplit(table) { decodedTable =>
```

Comment:
Technically this violates the contract for withRetry. We need all of the data on the GPU to be spillable and a table is not spillable.